### ISSUE TITLE
Uno Platform support

### ISSUE THREAD

**Issue by MartinZikmund** [verybadcat/CSharpMath#120]
**Is your feature request related to a problem? Please describe.**

Support for Uno Platform.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Difficulty: How difficult would it be? (Trivial, Very Easy, Easy, Moderate, Hard, Very Hard, Tedious, Backbreaking)**

Easy

**Additional context**

**Comment by Happypig375** [comment]
Nice. I also has this idea for quite a while now. Thank you!

**Comment by charlesroddie** [comment]
This already exists via SkiaSharp renderer.

**Comment by Happypig375** [comment]
This builds on top of the SkiaSharp renderer to integrate with Uno more nicely.

**Comment by Happypig375** [comment]
@charlesroddie A Uno front end would provide a MathView directly instead of having to call into drawing code, similar to CSharpMath.Avalonia and CSharpMath.Forms.

**Comment by charlesroddie** [comment]
This is a feature request and is not valid in its current form because the feature requested already exists (at least to our knowledge).

So we would need detail on what the current lack of support it, or assuming that there is already support, what improvement in support is needed.

**Comment by Happypig375** [comment]
There is already common code defining `MathView` and `TextView` shared by CSharpMath.Avalonia and CSharpMath.Forms, and the draft PR uses it to define these views for Platform Uno.

**Comment by MartinZikmund** [comment]
Yes, the goal is to add the controls that make it easy to use CSharpMath in Uno/UWP XAML

**Comment by Happypig375** [comment]
If that is the case, depending on SkiaSharp might not be the best approach. It would result in unexpected surprises to contributors. Would defining a new `MathPainter` and `TextPainter` that draw on `Windows.UI.Xaml.Controls.Canvas` be better? You can take a look at the Avalonia and SkiaSharp `MathPainter` and `TextPainter`s.

**Comment by MartinZikmund** [comment]
@Happypig375 the special SkiaSharp.Wasm dependency is only for the Uno project and only for WebAssembly target. It is also a normally available package on NuGet, so it shouldn't cause any issues.

**Comment by charlesroddie** [comment]
> But I will try if classic Canvas solution

Please don't! That would be massive duplication. We have existing duplication of logic for iOS for historical reasons (iOS was the only renderer at one point), and for Avalonia because that merged without proper assessment of costs and benefits.

**Comment by Happypig375** [comment]
>That would be massive duplication. [...] Avalonia because that was merged without proper assessment of costs and benefits.

Not really. The Avalonia front end is pretty small just like CSharpMath.SkiaSharp. Tests are written so that both projects are tested together.

**Comment by charlesroddie** [comment]
Actually yes the implementation there does seem a lot smaller than I remembered.

**Comment by Happypig375** [comment]
Because I refactored more common code with `CSharpMath.Forms` into `CSharpMath.Xaml`.

**Comment by Happypig375** [comment]
@MartinZikmund As discussed in #125, a rename from `CSharpMath.Uno` to `CSharpMath.UWPUno` will better stress that pure UWP users should also use this package. Thoughts?

**Comment by MartinZikmund** [comment]
Sounds good, I will rename the projects accordingly
